src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs

```diff
@@ -69,7 +74,10 @@ public static void ProcessLogFile(string filePath, string toolFormat, bool promp
 
             ITaskHandler taskHandler = taskStatusCenterService.PreRegister(taskHandlerOptions, taskProgressData);
 
-            taskHandler.RegisterTask(ProcessLogFileAsync(filePath, toolFormat, promptOnLogConversions, cleanErrors, openInEditor));
+            Task task = ProcessLogFileAsync(filePath, toolFormat, promptOnLogConversions, cleanErrors, openInEditor);
+            taskHandler.RegisterTask(task);
+
+            await task;
```

Comment:
Should we ConfigureAwait?

Reply:
Well, in reality you can just return the task and remove the "async" and "await" :)

But then again, I don't know what the JoinableTaskFactory.Run is going to do in that case.

---
In reply to: [530692825](https://github.com/microsoft/sarif-visualstudio-extension/pull/287#discussion_r530692825) [](ancestors = 530692825)

Reply:
I don't know why the threading analyzers didn't report this. It might be that if the `await` is the last statement in the function, it doesn't actually matter. In any case, I added `ConfigureAwait(false)`.

---
In reply to: [530694783](https://github.com/microsoft/sarif-visualstudio-extension/pull/287#discussion_r530694783) [](ancestors = 530694783,530692825)